plugins/device/init.go

```diff
@@ -32,12 +32,23 @@ func (p *Plugin) RestRoutes() route.Routes {
 			Version:      1,
 			RequestType:  utils.GetTypeString((*deviceapi.AddDeviceReq)(nil)),
 			ResponseType: utils.GetTypeString((*deviceapi.AddDeviceResp)(nil)),
-			HandlerFunc:  deviceAddHandler},
+			HandlerFunc:  deviceAddHandler,
+		},
+		route.Route{
+			Name:         "PeerEditGroup",
+			Method:       "POST",
+			Pattern:      "/peers/{peerid}/group",
+			Version:      1,
+			RequestType:  utils.GetTypeString((*deviceapi.PeerEditGroupReq)(nil)),
+			ResponseType: utils.GetTypeString((*deviceapi.PeerEditGroupResp)(nil)),
+			HandlerFunc:  peerEditGroupHandler,
+		},
 	}
 }
 
 // RegisterStepFuncs registers transaction step functions with
 // Glusterd Transaction framework
 func (p *Plugin) RegisterStepFuncs() {
 	transaction.RegisterStepFunc(txnPrepareDevice, "prepare-device")
+	transaction.RegisterStepFunc(txnPeerEditGroup, "peer-edit-group")
```

Comment:
Wait until the edit peer metadata PR gets in. You can re-use code (step function) from that PR here.